core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -438,59 +438,60 @@ sealed abstract class Parser1[+A] extends Parser[A] {
-    Parser.rep(this)
+  def rep0: Parser0[List[A]] =
+    Parser.rep0(this)
 
   /** Use this parser to parse at least `min` values (where `min >= 0`).
     *
     * If `min` is zero, this parser may succeed without consuming
     * input in the case where zero values are parsed. If `min` is
-    * known to be greater than zero, consider using `rep1(min)`
+    * known to be greater than zero, consider using `rep(min)`
     * instead.
     *
-    * Like `rep`, arresting failures in the underlying parser will
-    * result in an arresting failure. Unlike `rep`, this method may
+    * Like `rep0`, arresting failures in the underlying parser will
+    * result in an arresting failure. Unlike `rep0`, this method may
     * also return an arresting failure if it has not parsed at least
     * `min` values (but has consumed input).
     */
-  def rep(min: Int): Parser[List[A]] =
-    if (min == 0) rep
-    else rep1(min).map(_.toList)
+  def rep0(min: Int): Parser0[List[A]] =
+    if (min == 0) rep0
+    else rep(min).map(_.toList)
 
   /** Use this parser to parse one-or-more values.
     *
-    * This parser behaves like `rep`, except that it must produce at
+    * This parser behaves like `rep0`, except that it must produce at
     * least one value, and is guaranteed to consume input on successful
     * parses.
     */
-  def rep1: Parser1[NonEmptyList[A]] =
-    Parser.rep1(this, min = 1)
+  def rep: Parser[NonEmptyList[A]] =
+    Parser.rep(this, min = 1)
 
   /** Use this parser to parse at least `min` values (where `min >= 1`).
     *
-    * This method behaves likes `rep1`, except that if fewer than `min`
+    * This method behaves likes `rep`, except that if fewer than `min`
     * values are produced an arresting failure will be returned.
     */
-  def rep1(min: Int): Parser1[NonEmptyList[A]] =
-    Parser.rep1(this, min = min)
+  def rep(min: Int): Parser[NonEmptyList[A]] =
+    Parser.rep(this, min = min)
 
-  /** This method overrides `Parser#between` to refine the return type
+  /** This method overrides `Parser0#between` to refine the return type
     */
-  override def between(b: Parser[Any], c: Parser[Any]): Parser1[A] =
+  override def between(b: Parser0[Any], c: Parser0[Any]): Parser[A] =
     (b.void.with1 ~ (this ~ c.void)).map { case (_, (a, _)) => a }
 
-  /** This method overrides `Parser#surroundedBy` to refine the return type
+  /** This method overrides `Parser0#surroundedBy` to refine the return type
     */
-  override def surroundedBy(b: Parser[Any]): Parser1[A] =
+  override def surroundedBy(b: Parser0[Any]): Parser[A] =
     between(b, b)
 
-  /** This method overrides `Parser#soft` to refine the return type.
+  /** This method overrides `Parser0#soft` to refine the return type.
     */
-  override def soft: Parser.Soft10[A] =
-    new Parser.Soft10(this)
+  override def soft: Parser.Soft[A] =
+    new Parser.Soft(this)
 }
 
-object Parser extends ParserInstances {
+object Parser0 extends ParserInstances {}
```

Comment:
okay, I guess we need this because of the way implicit resolution works. Do we need the `{}`? Can we add a comment to explain to readers the design?